### ISSUE TITLE
Breaking changes on 2.42

### ISSUE THREAD

**Issue by ArcherEmiya05** [google/dagger#3428]
Currently reading [this ](https://github.com/google/dagger/releases/tag/dagger-2.42) release note, but cannot understand it completely. After update I am getting an error 
`cannot find implementation for com.sample.appdata.local.AppDatabase. AppDatabase_Impl does not exist`
and I think it is related to `@Component` stuff that was said in that release note.

This is the current setup looks like

```
class App : DaggerApplication() {

    override fun onCreate() {
        super.onCreate()

    }

    override fun applicationInjector(): AndroidInjector<out DaggerApplication> {
        val appComponent = DaggerAppComponent.builder().application(this).build()
        appComponent.inject(this)
        return appComponent
    }

}
```

```
@Singleton
@Component(
    modules = [
        AndroidSupportInjectionModule::class,
        AppModule::class,
        CommonModule::class,
        ViewModelFactoryModule::class,
        ActivityBuildersModule::class
    ]
)
interface AppComponent : AndroidInjector<App> {

    // Override the builder
    @Component.Builder
    interface Builder {

        @BindsInstance
        fun application(application: Application): Builder

        fun build(): AppComponent

    }

}
```

```
object AppModule {

    @Provides
    @Singleton
    @Named("AppName")
    fun provideAppName(application: Application): String {
        application.apply {
            return getString(R.string.app_name)
        }
    }

    @Provides
    @Singleton
    fun provideRoomDatabaseInstance(
        application: Application,
        @Named("AppName")
        appName: String,
        appJsonConverter: AppJsonConverter
    ): AppDatabase {
        return Room
            .databaseBuilder(application, AppDatabase::class.java, appName)
            .addTypeConverter(appJsonConverter)
            .fallbackToDestructiveMigration()
            .build()
    }

}
```

What exactly is the changes we need to perform to comply with the new version of Dagger? Thanks a lot

**Comment by bcorso** [comment]
Hi @ArcherEmiya05 ,

The issue in the release notes would result in a build failure, not a runtime failure, so I think this is something different.

Can you please post the entire stacktrace?

**Comment by ArcherEmiya05** [comment]
@bcorso it seems the issue is when using Kotlin 1.7.0. It keeps crashing with 1.7.0 but works fine with 1.6.20-M1. I think this issue has something to do with Kapt? Please correct me if I am wrong.

**Comment by ArcherEmiya05** [comment]
> I think we need more information to help. Do you have the full stacktrace? Also, can you clarify if the issue occurs when you upgrade Kotlin or Dagger? The title says when upgrading Dagger but you just mentioned it occurs when upgrading Kotlin?

Yes I think it has to do with Kotlin version 1.7.0 rather then the Dagger itself. Does Kapt work with the new Kotlin version by default?

**Comment by ArcherEmiya05** [comment]
> > I think we need more information to help. Do you have the full stacktrace? Also, can you clarify if the issue occurs when you upgrade Kotlin or Dagger? The title says when upgrading Dagger but you just mentioned it occurs when upgrading Kotlin?
> 
> Yes I think it has to do with Kotlin version 1.7.0 rather than the Dagger itself. Does Kapt work with the new Kotlin version by default?

Any update on this? Thanks

**Comment by Chang-Eric** [comment]
As far as I know kapt should still work with the new Kotlin version. If this is mostly about issues with upgrading Kotlin versions though, I'm not exactly sure how to help. That error message also seems to be coming from Room, not Dagger: https://cs.android.com/androidx/platform/frameworks/support/+/androidx-main:room/room-runtime/src/main/java/androidx/room/Room.kt;l=64;drc=c0e353885d721c9ae218a704905ec24028ddf335 I'm not sure this is really a Dagger question?
